src/txn/txn_recover.c

```diff
@@ -522,6 +532,15 @@ __wt_txn_recover(WT_SESSION_IMPL *session)
 done:	FLD_SET(conn->log_flags, WT_CONN_LOG_RECOVER_DONE);
 err:	WT_TRET(__recovery_free(&r));
 	__wt_free(session, config);
+
+	/*
+	 * Destroy the eviction threads that were started in support of
+	 * recovery.  They will be restarted once the lookaside table is
+	 * created.
+	 */
+	if (eviction_started)
+		WT_TRET(__wt_evict_destroy(session));
+
```

Comment:
I see what you mean about "avoiding ordering issues", I hadn't thought of it that way and I'd agree there are similar risks with both approaches.

The only solution I see is passing back a side-effect, that eviction was started inside recovery, and leaving the eviction-destroy outside of the recovery code, that guarantees anyone working in `__wt_connection_workers` will realize what's going on. But there are some obviously ugly things there, too.

So: if Alex and you have tried it the other way and this is better, I'm good to go.